pkg/util/test/tsdb.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package test
+
+import (
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/prometheus/tsdb/chunks"
+)
+
+type Series struct {
+	Labels  labels.Labels
+	Samples []Sample
+}
+
+type Sample struct {
+	TS        int64
+	Val       float64
+	Hist      *histogram.Histogram
+	FloatHist *histogram.FloatHistogram
+}
+
+func (s Sample) T() int64                      { return s.TS }
```

Comment:
Can we add a few comments here or otherwise document the semantics? Are float values, histograms, and float histograms mutually exclusive in a sample? Could we add constructors to enforce this?

Reply:
Each sample can only have one of each value type.
I think I would lean towards comments - I started this with a constructor, but it took up extra space in tests providing nils everywhere like `util_test.NewSample(0, 1.0, nil, nil)` vs. `util_test.Sample{TS: 0, Val: 1.0}`.

Reply:
The other option is something like this - fine with a panic here as it's a test util. Having the option to return an error would make all the tests a lot worse to write:
```golang
func NewSample(ts int64, val any) Sample {
	switch v := val.(type) {
	case float64:
		return Sample{TS: ts, Val: v}
	case *histogram.Histogram:
		return Sample{TS: ts, Hist: v}
	case *histogram.FloatHistogram:
		return Sample{TS: ts, FloatHist: v}
	default:
		panic("unsupported sample value type")
	}
}
```

Part of my problem with this is that type switches are not super cheap, and is an artificial thing to introduce into test paths that include benchmarking.